### PR TITLE
Add note on using index function for helm templating

### PR DESCRIPTION
**Pull request by rossf7**
Adding a note to the helm guide as suggested by @MarcelMue. See https://github.com/giantswarm/kubernetes-nginx-ingress-controller/pull/59#discussion_r217627887.

### CODE REVIEW

**Review by pipo02mix — APPROVED** [review]
really nice

File: helm/formatting_guidelines.md [hunk 1]
```diff
@@ -26,6 +26,14 @@ data:
   myvalue: "Hello World"
 ```
 
+When helm templating calls access values with underscores in the name you need
```
Comment (MarcelMue) [inline helm/formatting_guidelines.md]: `underscores` or `dashes`? Or possibly both?

**Review by MarcelMue — COMMENTED** [review]
Thanks a lot for adding this <3

**Review by MarcelMue — APPROVED** [review]
LGTM

**Review by kopiczko — APPROVED** [review]

**Comment by rossf7** [comment]
OK got 3 approvals. Going with this. Thanks Marcel for suggesting the docs update.

**Pull request merged**